src/Aspire.Hosting.Maui/MauiOtlpExtensions.cs

```diff
@@ -158,7 +158,7 @@ private static void ApplyOtlpConfigurationToPlatform<T>(
 
                 // Also remove the {RESOURCENAME}_{ENDPOINTNAME} format variable (e.g., MAUIAPP-OTLP_OTLP)
                 // The resource name keeps its case/dashes, endpoint name is uppercased
-                var directEndpointKey = $"{tunnelConfig.OtlpStub.Name.ToUpperInvariant()}_OTLP";
+                var directEndpointKey = $"{EnvironmentVariableNameEncoder.Encode(tunnelConfig.OtlpStub.Name).ToUpperInvariant()}_OTLP";
```

Comment:
?

Reply:
When the ENV is added for the endpoint it is encoded. The code here is trying to remove this value. So the logic needs to match. It's not "just" uppercase now, it's encoding and uppercase.

Reply:
Updated the comment and referenced where this is defined.